packages/block-events-stream/src/createBlockNumberStream.ts

```diff
@@ -0,0 +1,44 @@
+import { BehaviorSubject, filter, map } from "rxjs";
+import type { Block, BlockNumber, BlockTag, PublicClient } from "viem";
+import { createBlockStream } from "./createBlockStream";
+import { ReadonlyBehaviorSubject } from "./common";
+import { isNonPendingBlock } from "./isNonPendingBlock";
+
+// TODO: pass through viem's types, e.g. WatchBlocksParameters -> GetBlockReturnType
+
+export type CreateBlockNumberStreamOptions =
+  | {
+      publicClient: PublicClient;
+      blockTag: Omit<BlockTag, "pending">;
+      block$?: never;
+    }
+  | {
+      publicClient?: never;
+      blockTag?: never;
+      block$: ReadonlyBehaviorSubject<Block>;
+    };
+
+export async function createBlockNumberStream({
+  publicClient,
+  blockTag,
+  block$: initialBlock$,
+}: CreateBlockNumberStreamOptions): Promise<ReadonlyBehaviorSubject<BlockNumber>> {
+  const block$ = initialBlock$ ?? (await createBlockStream({ publicClient, blockTag: blockTag as BlockTag }));
```

Comment:
is it necessary to request the full block from the rpc here if we're only interested in the block number?

Reply:
`eth_blockNumber` doesn't support block tags and just returns the latest block number, so we have to request the block by block tag to get the number
(I already tried to refactor this to just use the block number RPC methods because they're cheaper, but ran into this issue)